os/arch/arm/src/amebad/amebad_enet.c

```diff
@@ -211,10 +215,16 @@ void up_netinitialize(void)
 		rtw_printf("Failed to register amebad netdev\n");
 	}
 #endif
+#ifndef CONFIG_VIRTUAL_BLE
 #if defined(CONFIG_AMEBAD_BLE) && defined(CONFIG_DRIVERS_BLE)
 	ameba_bm_dev_ble0 = bledev_register(&g_trble_drv_ops);
 	if (ameba_bm_dev_ble0 == NULL) {
 		rtw_printf("Failed to register amebad netdev\n");
 	}
 #endif
+#endif
+#ifdef CONFIG_VIRTUAL_BLE
+	vble_start();
```

Comment:
If user wants use both of h/w based driver & virtual driver, then this will not work properly.
It should have 2 drivers which has different path.

Reply:
Currently bledev_mgr supports only one bluetooth device driver.
            struct bledev *bledev_register(struct trble_ops *ops)
 It can not support more than one driver and hence either mock driver will work or real bluetooth driver will work.

Reply:
It's violence of our policy of driver....

Reply:
This behavior is consistent with the WIFI driver behavior

Reply:
> If user wants use both of h/w based driver & virtual driver, then this will not work properly. It should have 2 drivers which has different path.

There is a single entry point to driver. Virtual driver is for verification of Framework source code. It acts as a substitute of vendor driver. It helps in bugs segregation b/w Framework and vendor driver. I cannot see the use case of running both drivers at same time.